src/generators.rs

```diff
@@ -551,6 +551,421 @@ pub fn star_graph(
+        }
+    }
+    Ok(digraph::PyDiGraph {
+        graph,
+        node_removed: false,
+        check_cycle: false,
+        cycle_state: algo::DfsSpace::default(),
+    })
+}
+
+/// Generate an undirected grid graph.
+///
+/// :param int rows: The number of rows to generate the graph with.
+///     If specified, cols also need to be specified
+/// :param list cols: The number of rows to generate the graph with.
+///     If specified, rows also need to be specified. rows*cols
+///     defines the number of nodes in the graph
+/// :param list weights: A list of node weights. Nodes are filled row wise.
+///     If rows and cols are not specified, then a linear graph containing
+///     all the values in weights list is created.
+///     If number of nodes(rows*cols) is less than length of
+///     weights list, the trailing weights are ignored.
+///     If number of nodes(rows*cols) is greater than length of
+///     weights list, extra nodes with None weight are appended.
+///
+/// :returns: The generated grid graph
+/// :rtype: PyGraph
+/// :raises IndexError: If neither ``rows`` or ``cols`` and ``weights`` are
+///      specified
+///
+/// .. jupyter-execute::
+///
+///   import os
+///   import tempfile
+///
+///   import pydot
+///   from PIL import Image
+///
+///   import retworkx.generators
+///
+///   graph = retworkx.generators.grid_graph(2, 3)
+///   dot_str = graph.to_dot(
+///       lambda node: dict(
+///           color='black', fillcolor='lightblue', style='filled'))
+///   dot = pydot.graph_from_dot_data(dot_str)[0]
+///
+///   with tempfile.TemporaryDirectory() as tmpdirname:
+///       tmp_path = os.path.join(tmpdirname, 'dag.png')
+///       dot.write_png(tmp_path)
+///       image = Image.open(tmp_path)
+///       os.remove(tmp_path)
+///   image
+///
+#[pyfunction]
+#[text_signature = "(/, rows=None, cols=None, weights=None)"]
+pub fn grid_graph(
+    py: Python,
+    rows: Option<usize>,
+    cols: Option<usize>,
+    weights: Option<Vec<PyObject>>,
+) -> PyResult<graph::PyGraph> {
+    let mut graph = StableUnGraph::<PyObject, PyObject>::default();
+    if weights.is_none() && (rows.is_none() || cols.is_none()) {
+        return Err(PyIndexError::new_err(
+            "dimensions and weights list not specified",
+        ));
+    }
+
+    let mut rowlen = match rows {
+        Some(rows) => rows,
+        None => 0,
+    };
+
+    let mut collen = match cols {
+        Some(cols) => cols,
+        None => 0,
+    };
+
+    let mut num_nodes = rowlen * collen;
+
+    let nodes: Vec<NodeIndex> = match weights {
+        Some(weights) => {
+            let mut node_list: Vec<NodeIndex> = Vec::new();
+            if num_nodes < weights.len() && rowlen == 0 {
+                collen = weights.len();
+                rowlen = 1;
+                num_nodes = collen;
+            }
+
+            let mut node_cnt = num_nodes;
+
+            for weight in weights {
+                if node_cnt <= 0 {
```

Comment:
I ran `cargo clippy` on the PR locally and it flagged this. Since `node_cnt` is a usize it can't ever be less than 0, so they suggested replacing this with:

```suggestion
                if node_cnt == 0 {
```